### ISSUE TITLE
Tracking issue: OpenHarmony port

### ISSUE THREAD

**Issue by jschwe** [servo/servo#32644]
As of #32594 servoshell now supports OpenHarmony, and the respective build is also tested in CI. 
However, there are also still a number of issues, and this tracking issues serves the purpose of giving an overview over the remaining issues. 

- [ ] Update the servo documentation with instructions on how to build servo for OpenHarmony.
- [ ] Redirect stdout/stderr to hilog (see android logcat redirection, perhaps we can unify the implementation)
- [ ] Improved font support. servo should parse the fontconfig.json file, and improve font-fallback logic on ohos.
- [ ] WebGL: WebGL currently needs to be turned off, since offscreen surface support for ohos is not implemented
    yet in surfman.
- [ ]  IPC-channel cross-process support. OH 4.1+ adds `shm_open` support, but at runtime the shm_open / memfd calls in ipc_channel fails. Needs investigation on what the error code is, and what we can do about it.
- [ ] Vsync: Probably updates should be batched and processed based on the vertical synchronization signal. This is likely a pre-requiste for Fling support. Without fling, the input touch event notifications probably serve as a poor-mans vsync signal when scrolling.
- [ ] Add a back button
- [ ] Input Events / Gestures:
  - [ ] Fling support (note: On android fling only seems to work in the "up" direction, but not downwards, so perhaps we can do something shared here)
  - [ ] Have some "scroll-mode", which disables navigating to links, when the user lifts their finger while over a link, after scrolling.
  - [ ] Keyboard events (probably low priority for now)
- [ ] Support configuring initialization parameters via some method, e.g. from a configuration file.
- [ ] Reading files on OH: Apps are sandboxed. `Apps` support shipping files via `rawfile` and `resfile` methods. `rawfile` requires that code interacts with the file using specialised rawfile APIs. Resfile files should appear in the sandboxed area, after the app is installed, but that didn't seem to work for me on first try.
- [ ] Move the [ArkTS ServoDemo app](https://github.com/jschwe/ServoDemo) to the `support` folder in servo and integrate the `.hap` build into `mach` or a different tool, which could wrap the whole ohos app build , sign and flash process.
- [ ] Run the Ohos app in the emulator in CI. This is blocked on either the HarmonyOS NEXT emulator  becoming publicly available, or figuring out how to run pure OpenHarmony on qemu.

**Comment by Taym95** [comment]
@jschwe I may be able to help here. How do you test your changes? Do you run OpenHarmony in an emulator?

**Comment by jschwe** [comment]
@Taym95 Thanks for the interest! I mostly test on real hardware, but I've also used the HarmonyOS emulator to test my changes. However, that emulator is not publicly available yet - but we hope the emulator will be made generally available soon. I will post an update here, once that is the case.

**Comment by mrobinson** [comment]
@jschwe What else is left for this issue? It seems like the OHOS port has come a long way!
